### PR TITLE
[Model] Rename MiniCPMVQwen2 to MiniCPMV2.6

### LINKED ISSUE #7267: [New Model]:Is MiniCPM-V-2_6 supported?
### The model to consider.

MiniCPM-V-2_6

### The closest model vllm already supports.

_No response_

### What's your difficulty of supporting the model you want?

_No response_

### PR DESCRIPTION
**Pull request by jeejeelee**
I have completed the following modification:

- Rename MiniCPMVQwen2 to MiniCPMV2.6
- Slightly modified some logic
- Add MiniCPMV2.6 to list of supported models (FIX #7267)

ping @ywang96  @DarkLight1337 @HwwwwwwwH

### CODE REVIEW

File: vllm/model_executor/models/minicpmv.py [hunk 1]
```diff
@@ -1007,20 +1003,11 @@ def __new__(
...
     ):
-        if not hasattr(config, "version"):
-            if config.hidden_size == 2304 and config.query_num == 64:
-                version = (2, 0)
-            else:
-                version = (2, 5)
-        else:
-            version = str(config.version).split(".")
-            version = tuple([int(x) for x in version])
+        version = str(config.version).split(".")
```
Comment (DarkLight1337) [inline vllm/model_executor/models/minicpmv.py:1006]: Has the config for v2.0 been updated yet?

File: vllm/model_executor/models/minicpmv.py [hunk 2]
```diff
@@ -1007,20 +1003,11 @@ def __new__(
...
         # Dispatch class based on version
-        if version == (2, 0):
-            instance_class = MiniCPMV2
-        elif version == (2, 5):
-            instance_class = MiniCPMV2_5
-        else:
-            instance_class = MiniCPMVQwen2
+        instance_class = _SUPPORT_VERSION.get(version, None)
+        if instance_class is None:
+            raise NotImplementedError
```
Comment (DarkLight1337) [inline vllm/model_executor/models/minicpmv.py:1011]: A more informative error message would be nice.

**Comment by DarkLight1337** [comment]
Could you also update the docs to include this in the list of supported models?

File: vllm/model_executor/models/minicpmv.py (hunk 1 above)
Comment (jeejeelee) [reply vllm/model_executor/models/minicpmv.py:1006]: Not yet, my mistake

File: vllm/model_executor/models/minicpmv.py [hunk 3]
```diff
@@ -216,16 +216,7 @@ def __init__(
...
         trunc_normal_(self.query, std=0.02)
-
-        if kv_dim is not None and kv_dim != embed_dim:
-            self.kv_proj = ReplicatedLinear(kv_dim, embed_dim, bias=False)
-        else:
-            # Maintain the same return value with ReplicatedLinear.forward
-            self.kv_proj = lambda *args, **kwargs: (
-                nn.Identity()(*args, **kwargs),
-                None,
-            )
```
Comment (DarkLight1337) [inline vllm/model_executor/models/minicpmv.py:227]: Could you explain why you made this change?

Comment (jeejeelee) [reply vllm/model_executor/models/minicpmv.py:227]: If I haven't tested incorrectly, in models 2.0, 2.5, and 2.6, this layer is an instance of ReplicatedLinear.

Comment (DarkLight1337) [reply vllm/model_executor/models/minicpmv.py:227]: I think we should not assume that `kv_dim != embed_dim`. Even if this is true for the official HF repo, those who train a custom version of the model might have different configs.

Comment (jeejeelee) [reply vllm/model_executor/models/minicpmv.py:227]: How should we modify it?

Comment (DarkLight1337) [reply vllm/model_executor/models/minicpmv.py:227]: I think the original logic should work fine.

Comment (jeejeelee) [reply vllm/model_executor/models/minicpmv.py:227]: Fine, I will revert this snippet

**Review by DarkLight1337 — APPROVED** [review]

**Comment by HwwwwwwwH** [comment]
Sry for late. Thank you for your work! There're something still need attention. The example of `minicpmv` should be extended to contain three versions. And three versions of `minicpmv` need different `stop_token_ids`. So if the user run `example/offline_inference_vision_language.py`, for example, for V2.5, they might got many `<|eot|>` at the end of outputs.
So the logic of `offline_inference_vision_language.py` many need a slight modification.

**Comment by jeejeelee** [comment]
> Sry for late. Thank you for your work! There're something still need attention. The example of `minicpmv` should be extended to contain three versions in `example/offline_inference_vision_language.py`. And three versions of `minicpmv` need different `stop_token_ids`. So if the user run `example/offline_inference_vision_language.py`, for example, for V2.5, they might got many `<|eot|>` at the end of outputs. So the logic of `offline_inference_vision_language.py` many need a slight modification.

Ok, is that right？
```python

  # 2.0
  stop_token_ids = [tokenizer.eos_id]
  # 2.5
  # stop_token_ids = [tokenizer.eos_id, tokenizer.eot_id]
  #2.6
  # stop_tokens = ['<|im_end|>', '<|endoftext|>']
  # stop_token_ids = [tokenizer.convert_tokens_to_ids(i) for i in stop_tokens]
```

**Comment by HwwwwwwwH** [comment]
> 
> Ok, is that right？
> 
> ```python
>   # 2.0
>   stop_token_ids = [tokenizer.eos_id]
>   # 2.5
>   # stop_token_ids = [tokenizer.eos_id, tokenizer.eot_id]
>   #2.6
>   # stop_tokens = ['<|im_end|>', '<|endoftext|>']
>   # stop_token_ids = [tokenizer.convert_tokens_to_ids(i) for i in stop_tokens]
> ```
Yes

**Comment by jeejeelee** [comment]
> > Ok, is that right？
> > ```python
> >   # 2.0
> >   stop_token_ids = [tokenizer.eos_id]
> >   # 2.5
> >   # stop_token_ids = [tokenizer.eos_id, tokenizer.eot_id]
> >   #2.6
> >   # stop_tokens = ['<|im_end|>', '<|endoftext|>']
> >   # stop_token_ids = [tokenizer.convert_tokens_to_ids(i) for i in stop_tokens]
> > ```
> 
> Yes

@DarkLight1337   I am not sure if I can add these in [vl example](https://github.com/vllm-project/vllm/blob/v0.5.4/examples/offline_inference_vision_language.py)

**Comment by DarkLight1337** [comment]
Feel free to update the example!

**Comment by DarkLight1337** [comment]
@HwwwwwwwH can you try running the example on your end? I'm outside right now.

**Comment by jeejeelee** [comment]
@HwwwwwwwH  I'd like to know why you chose to use the `LlamaModel` rather than `LlamaForCausalLM` as the language model for minicpm2.5. This approach actually changes the model hierarchy, which is not conducive to the support of LoRA.

**Comment by HwwwwwwwH** [comment]
> @HwwwwwwwH can you try running the example on your end? I'm outside right now.


> @HwwwwwwwH can you try running the example on your end? I'm outside right now.

I'm running it now.

**Comment by HwwwwwwwH** [comment]
> @HwwwwwwwH I'd like to know why you chose to use the `LlamaModel` rather than `LlamaForCausalLM` as the language model for minicpm2.5. This approach actually changes the model hierarchy, which is not conducive to the support of LoRA.

emmmm, at first I used `xxxCausalLM`, but then I found that all of the `CausalLM` deleted the `inputs_embeds` parameters while kept it in `xxModel` . And using `xxModel` is what other VLMs do, so I followed this way.

**Comment by HwwwwwwwH** [comment]
I got this error when running `gguf`. And I don't see any dependency in requirements files. Maybe you need add to this?
```
Traceback (most recent call last):
  File "/data1/a1/vllm_e/examples/offline_inference_vision_language.py", line 10, in <module>
    from vllm import LLM, SamplingParams
  File "/data1/a1/vllm_e/vllm/__init__.py", line 3, in <module>
    from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
  File "/data1/a1/vllm_e/vllm/engine/arg_utils.py", line 7, in <module>
    from vllm.config import (CacheConfig, DecodingConfig, DeviceConfig,
  File "/data1/a1/vllm_e/vllm/config.py", line 11, in <module>
    from vllm.model_executor.layers.quantization import QUANTIZATION_METHODS
  File "/data1/a1/vllm_e/vllm/model_executor/layers/quantization/__init__.py", line 16, in <module>
    from vllm.model_executor.layers.quantization.gguf import GGUFConfig
  File "/data1/a1/vllm_e/vllm/model_executor/layers/quantization/gguf.py", line 3, in <module>
    import gguf
ModuleNotFoundError: No module named 'gguf'
```

**Comment by DarkLight1337** [comment]
> I got this error when running `gguf`. And I don't see any dependency in requirements files. Maybe you need add to this?
> ```
> Traceback (most recent call last):
>   File "/data1/a1/vllm_e/examples/offline_inference_vision_language.py", line 10, in <module>
>     from vllm import LLM, SamplingParams
>   File "/data1/a1/vllm_e/vllm/__init__.py", line 3, in <module>
>     from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
>   File "/data1/a1/vllm_e/vllm/engine/arg_utils.py", line 7, in <module>
>     from vllm.config import (CacheConfig, DecodingConfig, DeviceConfig,
>   File "/data1/a1/vllm_e/vllm/config.py", line 11, in <module>
>     from vllm.model_executor.layers.quantization import QUANTIZATION_METHODS
>   File "/data1/a1/vllm_e/vllm/model_executor/layers/quantization/__init__.py", line 16, in <module>
>     from vllm.model_executor.layers.quantization.gguf import GGUFConfig
>   File "/data1/a1/vllm_e/vllm/model_executor/layers/quantization/gguf.py", line 3, in <module>
>     import gguf
> ModuleNotFoundError: No module named 'gguf'
> ```

It should be in the latest `requirements.txt`. Maybe need to sync this branch with `main`.

**Comment by DarkLight1337** [comment]
> > @HwwwwwwwH I'd like to know why you chose to use the `LlamaModel` rather than `LlamaForCausalLM` as the language model for minicpm2.5. This approach actually changes the model hierarchy, which is not conducive to the support of LoRA.
> 
> emmmm, at first I used `xxxCausalLM`, but then I found that all of the `CausalLM` deleted the `inputs_embeds` parameters while kept it in `xxModel` . And using `xxModel` is what other VLMs do, so I followed this way.

Yeah, the current VLMs all use the `*Model` class rather than `*ForCausalLM`.

**Comment by HwwwwwwwH** [comment]
> @HwwwwwwwH can you try running the example on your end? I'm outside right now.

It's ok with the examples.

**Review by DarkLight1337 — APPROVED** [review]

**Comment by jeejeelee** [comment]
> > > @HwwwwwwwH I'd like to know why you chose to use the `LlamaModel` rather than `LlamaForCausalLM` as the language model for minicpm2.5. This approach actually changes the model hierarchy, which is not conducive to the support of LoRA.
> > 
> > 
> > emmmm, at first I used `xxxCausalLM`, but then I found that all of the `CausalLM` deleted the `inputs_embeds` parameters while kept it in `xxModel` . And using `xxModel` is what other VLMs do, so I followed this way.
> 
> Yeah, the existing VLMs use the `*Model` class rather than `*ForCausalLM`. There have been efforts to change this though to use @Isotr0py 's recipe for loading inner models. (See #7153)

Sorry for deleting my previous comments by mistake. I will read  these codes, thank you.

**Pull request merged**